tapgarden/custodian.go

```diff
@@ -828,6 +826,35 @@ func (c *Custodian) mapProofToEvent(p proof.Blob) error {
 	return nil
 }
 
+// assertProofInLocalArchive checks if the proof is already in the local proof
+// archive. If it isn't, it is imported now.
+func (c *Custodian) assertProofInLocalArchive(p *proof.AnnotatedProof) error {
+	ctxt, cancel := c.WithCtxQuit()
+	defer cancel()
+
+	haveProof, err := c.cfg.ProofArchive.HasProof(ctxt, p.Locator)
+	if err != nil {
+		return fmt.Errorf("error checking if proof is available: %w",
+			err)
+	}
+
+	// We don't have the proof yet, or not in all backends, so we
+	// need to import it now.
+	if !haveProof {
+		headerVerifier := GenHeaderVerifier(ctxt, c.cfg.ChainBridge)
+		err = c.cfg.ProofArchive.ImportProofs(
+			ctxt, headerVerifier, c.cfg.GroupVerifier, false, p,
+		)
+		if err != nil {
+			log.Errorf("ERROOOORRR: %v", err)
```

Comment:
Are we leaving this in as-is? 😅

Reply:
Can push a commit to resolve on our timezone before merging.